File: monitoring-server/MonitoringServer/Models/Monitor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace MonitoringService.Models
{
    /// <summary>
    /// Represents monitor for particular equipment instance.
    /// Each instance of monitor can track single equipment 
    /// instance but specified set of instance params.
    /// </summary>
    public class Monitor
    {
        /// <summary>
        /// Internal monitor identidier.
        /// </summary>
        public long Id { get; set; }
        /// <summary>
        /// External monitor identifier.
        /// </summary>
        public Guid MonitorUid { get; set; }
        /// <summary>
        /// Monitor name.
        /// </summary>
        public string Name { get; set; }
        /// <summary>
        /// Equipment instance external identifier that monitor belongs to.
        /// </summary>
        public Guid EquipmentUid { get; set; }
        /// <summary>
        /// Equipment model external identifier that monitor belongs to.
        /// </summary>
        public Guid EquipmentModelUid { get; set; }
        /// <summary>
        /// Navigation property to all tracked params.
        /// </summary>
        public virtual List<MonitorParam> MonitorParams { get; set; }
    }
}

File: monitoring-server/monitoring-service/src/main/scala/com/qwerfah/monitoring/models/Monitor.scala
package com.qwerfah.monitoring.models

import com.qwerfah.common.Uid

/** Represent equipment instance state monitor.
  * @param id
  *   Internal monitor identifier.
  * @param uid
  *   External monitor identifier.
  * @param instanceUid
  *   Related equipment instance id.
  * @param name
  *   Monitor name.
  * @param description
  *   Monitor description (optional).
  */
final case class Monitor(
  id: Option[Int],
  uid: Uid,
  instanceUid: Uid,
  name: String,
  description: Option[String]
)
